modules/cli/src/main/java/org/apache/ignite/internal/cli/core/repl/completer/DynamicCompleterActivationPoint.java

```diff
@@ -55,10 +57,22 @@ public void activateDynamicCompleter(DynamicCompleterRegistry registry) {
         );
         registry.register(
                 CompleterConf.builder()
-                        .command("node", "config", "show")
-                        .command("node", "config", "update").build(),
+                        .command("node", "config", "show").build(),
                 nodeConfigDynamicCompleterFactory
         );
+
+        registry.register(
+                CompleterConf.builder()
+                        .command("node", "config", "update")
+                        .filter((unused, candidates) -> {
```

Comment:
This is a good idea to define a filter as a lambda right here but this approach has an issue: it is not covered with unit tests. You can define a separate class like `NodeConfigUpdateFilter` and test it with unite tests.